packages/lit-html/src/static.ts

```diff
@@ -0,0 +1,109 @@
+/**
+ * @license
+ * Copyright (c) 2020 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+import {html as coreHtml, svg as coreSvg, TemplateResult} from './lit-html.js';
+
+/**
+ * Wraps a string so that it behaves like part of the static template
+ * strings instead of a dynamic value.
+ *
+ * This is a very unsafe operation and may break templates if changes
+ * the structure of a template. Do not pass user input to this function
+ * without sanitizing it.
+ *
+ * Static values can be changed, but they will cause a complete re-render
+ * since they effectively create a new template.
+ */
+export const unsafeStatic = (value: string) => ({
+  _$litStatic$: value,
+});
+
+type StaticValue = ReturnType<typeof unsafeStatic>;
+
+const stringsCache = new Map<string, TemplateStringsArray>();
+
+/**
+ * Generates a template literal tag function that returns a TemplateResult with
+ * the given result type.
+ */
+const tag = (coreTag: typeof coreHtml) => (
+  strings: TemplateStringsArray,
+  ...values: unknown[]
+): TemplateResult => {
+  let key = strings[0];
+  for (let i = 0; i < strings.length; i++) {
+    if ((values[i] as StaticValue)?._$litStatic$ !== undefined) {
+      key += (values[i] as StaticValue)._$litStatic$;
+    } else {
+      key += '${lit-html-joiner}';
```

Comment:
Maybe make this shorter? Any reason it looks like a template expression?

Reply:
Maybe add a comment about why this is included in the key? It's so that the values length is part of the key and helps with uniqueness, yeah?

Reply:
I was first inspired by @aomarks using `${` as a joiner in @lit/localize on the idea that it can't be used in a template string without escaping, but then realized that the non-raw string value would still contain `${` even if escaped. It can be anything unlikely to appear in the template.